Show relationships in included resource objects

ethanresnick opened the pull request:
Even though the spec text says that the objects in `included` are
resource objects (and so can have relationships), I’ve had a couple
people tell me they weren’t sure whether this was allowed because we
didn’t have an example showing it. This aims to rectify that.

ethanresnick committed:
Demonstrate relationships in included resource objects

Even though the spec text says that the objects in `included` are
resource objects (and so can have relationships), I’ve had a couple
people tell me they weren’t sure whether this was allowed because we
didn’t have an example showing it. This aims to rectify that.

Even though the spec text says that the objects in `included` are
resource objects (and so can have relationships), I’ve had a couple
people tell me they weren’t sure whether this was allowed because we
didn’t have an example showing it. This aims to rectify that.

dgeb commented:
Thanks, @ethanresnick. I've had this same question come up, so this is a great idea. Would you mind also updating the home page example with the same changes (since it's the same example)?

ethanresnick committed:
Change example type from posts to articles

For consistency with the example on the specification page and to
prevent confusion with POST.

ethanresnick commented:
@dgeb Done! I also changed the `type` on the homepage example to match the one on the specification page.

ethanresnick commented:
@dgeb @tkellen this good to merge now?

tkellen commented:
:+1:

ethanresnick commented:
cool beans
